File: src/main/java/org/goafabric/personservice/logic/PersonMapper.java
package org.goafabric.personservice.logic;

import jakarta.data.page.Page;
import org.goafabric.personservice.controller.dto.Person;
import org.goafabric.personservice.persistence.entity.PersonEo;
import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;

import java.util.List;


@Mapper(componentModel = "jakarta", unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface PersonMapper {
    Person map(PersonEo value);

    PersonEo map(Person value);

    List<Person> map(List<PersonEo> value);

    List<Person> map(Page<PersonEo> value);
}

File: src/main/kotlin/org/goafabric/personservice/logic/PersonMapper.kt
package org.goafabric.personservice.logic

import io.mcarle.konvert.api.Konfig

import io.mcarle.konvert.api.Konverter
import io.mcarle.konvert.injector.cdi.KApplicationScoped

import org.goafabric.personservice.controller.dto.Address
import org.goafabric.personservice.controller.dto.Person
import org.goafabric.personservice.persistence.entity.AddressEo
import org.goafabric.personservice.persistence.entity.PersonEo

@Konverter([Konfig(key = "konvert.enforce-not-null", value = "true")])
@KApplicationScoped
interface PersonMapper {
    fun map(value: PersonEo): Person

    fun map(value: Person): PersonEo

    fun map(value: List<PersonEo>): List<Person>

    fun map(value: AddressEo): Address

    fun map(value: Address): AddressEo


}
